test/reserves.test.ts

```diff
@@ -0,0 +1,133 @@
+import { ONE_ALPH, web3 } from '@alephium/web3'
+import { getSigner } from '@alephium/web3-test'
+import { PrivateKeyWallet } from '@alephium/web3-wallet'
+import { deployInvariant, newFeeTier, newPoolKey } from '../src/utils'
+import { MaxSqrtPrice, PercentageScale } from '../src/consts'
+import {
+  getPool,
+  initPool,
+  initFeeTier,
+  initTokensXY,
+  initPosition,
+  withdrawTokens
+} from '../src/testUtils'
+import { InvariantInstance } from '../artifacts/ts'
+import { FeeTier } from '../artifacts/ts/types'
+import {
+  getBasicFeeTickSpacing,
+  initBasicPool,
+  initBasicPosition,
+  initBasicSwap
+} from '../src/snippets'
+
+web3.setCurrentNodeProvider('http://127.0.0.1:22973')
+let admin: PrivateKeyWallet
+let positionOwner: PrivateKeyWallet
+let swapper: PrivateKeyWallet
+let invariant: InvariantInstance
+describe('manage multiple tokens', () => {
```

Comment:
Is there a case where you create a pool and use an existing reserve instead of creating a new one because the reserve for these tokens already exists? For one or both existing tokens - maybe the same pair with a different fee tier?

Reply:
There is a case, where pool is created X/Y and then X/Z. The X is using same reserve on both pools. Creating pool wtih same pair on different fee tier is not tested